### PR TITLE
#643 Part 1: Upstream changes required for transcriptomic overview figure

### PR DESCRIPTION
**Pull request by jaclyn-taroni**
### Purpose/implementation Section

Generating the transcriptomic overview in #643 requires several upstream modules to be run:

* `transcriptomic-dimension-reduction`
* `collapse-rnaseq`
* `gene-set-enrichment-analysis`
* `immune-deconv`

I found that this application required a few changes to upstream modules, so I decided to file a pull request before I completed #643. I've held off on documentation around this figure for now.

#### What was your approach?

See directions for reviews for more information.

#### What GitHub issue does your pull request address?

Begins to address #643

#### Which areas should receive a particularly close look?

I would like to direct your attention to the following:

* The changes to `figures/generate-figures.sh` include the upstream changes required before putting together the figure outlined in #643. This probably needs the _closest_ look.
* I altered `analyses/gene-set-enrichment-analysis/01-conduct-gsea-analysis.R` and the shell script for that module `analyses/gene-set-enrichment-analysis/run-gsea.sh` such that the assumptions about where input and output files are located are relaxed.
* Because we don't include CIBERSORT in this repository and I don't have access to it, I am using xCell and MCP counter for `immune-deconv`. This uncovered an issue with how we were handle this case in `analyses/immune-deconv/01-immune-deconv.R`. 

Any changes in the TSV files related to `transcriptomic-dimension-reduction` appear to be due to #541 

#### Is there anything that you want to discuss further?

I am ignoring the RData output of `analyses/immune-deconv/01-immune-deconv.R` that doesn't use CIBERSORT/is specifically for generating figures - what do folks think of this approach?

#### Reproducibility Checklist

For the altered modules.

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

**Comment by sjspielman** [comment]
Some initial questions/comments before I dive in more tomorrow:

+  Inconsistent use of `--vanilla` flag when running R scripts that generate figures. Do we care? I'm not sure if it matters, but in case it does, we should do it for all or none.

+ I can see `figures/generate-figures.sh` becoming rather large and potentially unwieldy - especially since a lot of the code isn't explicitly generating a figure but rather prepping data, as is the case in this PR. Is there a good reason *not* to break this down into other files within `figures/`? Smaller shells scripts for cohesive groups of figures which employ the same data, and then call all those scripts from `generate-figures.sh`?

**Comment by jaclyn-taroni** [comment]
We have some ideas about how to improve how we generate figures, namely #630. We’ve chosen to go unwieldy as a first pass.

**Comment by jaclyn-taroni** [comment]
> Inconsistent use of --vanilla flag when running R scripts that generate figures. Do we care? I'm not sure if it matters, but in case it does, we should do it for all or none.

I don't think we care so much, but I added it in.

**Comment by jaclyn-taroni** [comment]
@sjspielman I think this is ready for another look (provided the immune deconvolution step passes CI). I made some changes such that the second method for `analyses/immune-deconv/01-immune-deconv.R` can be NULL, so when we generate figures it will use xCell.

File: analyses/immune-deconv/01-immune-deconv.R [hunk 1]
```diff
@@ -67,37 +69,36 @@ clin <- read.delim(clin.file, stringsAsFactors = F)
...
     inner_join(clin.sub, by = c("sample" = "Kids_First_Biospecimen_ID"))
-  
+
   return(res)
 }
 
 # Deconvolute using xCell and the second chosen method
-# these two methods have the max number of immune cell types
-deconv.method <- c("xcell", deconv.method)
+deconv.method <- unique(c("xcell", deconv.method))
```
Comment (sjspielman) [inline analyses/immune-deconv/01-immune-deconv.R:95]: Good call here, makes sense!!!

**Pull request merged**